### PR TITLE
Display 'add-comment' form only if user is logged in

### PR DESCRIPTION
**Pull request by mariom3**
Add the front end to display appropriate form depending on whether the user is logged in or not

### CODE REVIEW

File: portfolio/src/main/webapp/script.js [hunk 1]
```diff
@@ -36,4 +35,69 @@ function createCommentElement(userName, comment) {
...
+      // login to comment
+      const loginButton = document.createElement('button');
+      loginButton.innerText = 'LOGIN TO ADD A COMMENT';
+      loginButton.setAttribute('class', 'btn btn-outline-dark custom-btn title-font');
+      addCommentElement.appendChild(loginButton);
+    }
+  });
+}
+
+function createCommentForm() {
```
Comment (sbahaud) [inline portfolio/src/main/webapp/script.js]: You could do display:none or other approaches to hide for  form element. Then this function can just toggle that status
https://allyjs.io/tutorials/hiding-elements.html

File: portfolio/src/main/webapp/script.js [hunk 2]
```diff
@@ -36,4 +35,69 @@ function createCommentElement(userName, comment) {
   divElement.innerText = userName + ': ';
   divElement.appendChild(pElement);
   return divElement;
-}
\ No newline at end of file
+}
+
+function getLoggedInStatus() {
+  fetch('/data').then(response => response.json()).then((json) => {
```
Comment (sbahaud) [inline portfolio/src/main/webapp/script.js:40]: Move to a different endpoint. That keeps the data endpoint for just fetching the comments.

**Review by sbahaud — COMMENTED** [review]
Left some comments

File: portfolio/src/main/webapp/script.js (hunk 1 above)
Comment (mariom3) [reply portfolio/src/main/webapp/script.js]: Done: 2dfab35 
This is useful. Thank you.

File: portfolio/src/main/webapp/script.js (hunk 2 above)
Comment (mariom3) [reply portfolio/src/main/webapp/script.js:40]: Will be addressed when I refactor code as an update to PR #14

File: portfolio/src/main/webapp/css/aboutme.css [hunk 3]
```diff
@@ -61,6 +56,17 @@ body {
   padding-right: 1em;
 }
 
+#login-logout {
+  padding: .3em;
+  background: red;
```
Comment (sbahaud) [inline portfolio/src/main/webapp/css/aboutme.css]: nit: Order alphabetically

File: portfolio/src/main/webapp/script.js [hunk 4]
```diff
@@ -36,4 +34,35 @@ function createCommentElement(userName, comment) {
...
+    const loggedIn = json.loggedInStatus;
+
+    if (loggedIn){
+      const pElement = document.getElementById('display-user');
+      pElement.style.display = "block";
+      pElement.textContent = "Logged in as: " + json.userEmail;
+      document.getElementById('logout-link').href = json.logoutUrl;
+      document.getElementById('comment-form').style.display = "block";
+    }else
```
Comment (sbahaud) [inline portfolio/src/main/webapp/script.js]: nit: fix formatting - space after } and before {

File: portfolio/src/main/webapp/script.js [hunk 5]
```diff
@@ -36,4 +34,35 @@ function createCommentElement(userName, comment) {
...
   return divElement;
-}
\ No newline at end of file
+}
+
+function getLoggedInStatus() {
+  fetch('/data').then(response => response.json()).then((json) => {
+    const loggedIn = json.loggedInStatus;
+
+    if (loggedIn){
```
Comment (sbahaud) [inline portfolio/src/main/webapp/script.js]: nit: fix formatting, space before '{' 
loggedIn) {

File: portfolio/src/main/webapp/script.js [hunk 6]
```diff
@@ -36,4 +34,35 @@ function createCommentElement(userName, comment) {
...
+      const pElement = document.getElementById('display-user');
+      pElement.style.display = "block";
```
Comment (sbahaud) [inline portfolio/src/main/webapp/script.js]: To make the code cleaner, you could add a method like displayElement(element, show)

if (show) {
  element.style.display = "block"
} else {
  element.style.display = "none";
}

File: portfolio/src/main/webapp/script.js [hunk 7]
```diff
@@ -36,4 +34,35 @@ function createCommentElement(userName, comment) {
...
+      pElement.textContent = "Logged in as: " + json.userEmail;
```
Comment (sbahaud) [inline portfolio/src/main/webapp/script.js:46]: Maybe rename to displayName from userEmail

File: portfolio/src/main/webapp/script.js [hunk 8]
```diff
@@ -36,4 +34,35 @@ function createCommentElement(userName, comment) {
...
+      document.getElementById('logout-link').href = json.logoutUrl;
+      document.getElementById('comment-form').style.display = "block";
+    }else
+      createLoginButton(json.loginUrl);
+  });
+}
+
+function createLoginButton(loginUrl) {
```
Comment (sbahaud) [inline portfolio/src/main/webapp/script.js]: Can you do the button creation etc in html and only have this method set the href?

Comment (sbahaud) [reply portfolio/src/main/webapp/script.js]: Another option is to redirect them to a login page and not have a button in the main page.

**Review by sbahaud — COMMENTED** [review]
Added some more comments

File: portfolio/src/main/webapp/script.js (hunk 8 above)
Comment (mariom3) [reply portfolio/src/main/webapp/script.js]: Done: 60eacb3 
- add button in html

File: portfolio/src/main/webapp/script.js (hunk 6 above)
Comment (mariom3) [reply portfolio/src/main/webapp/script.js]: Done: dbe9632

File: portfolio/src/main/webapp/css/aboutme.css (hunk 3 above)
Comment (mariom3) [reply portfolio/src/main/webapp/css/aboutme.css]: Done: d8be3be

File: portfolio/src/main/webapp/script.js (hunk 5 above)
Comment (mariom3) [reply portfolio/src/main/webapp/script.js]: Done: dbe9632

File: portfolio/src/main/webapp/script.js (hunk 4 above)
Comment (mariom3) [reply portfolio/src/main/webapp/script.js]: Done: dbe9632

File: portfolio/src/main/webapp/script.js (hunk 7 above)
Comment (mariom3) [reply portfolio/src/main/webapp/script.js:46]: Will do when I refactor code as an update to PR #14

**Review by sbahaud — APPROVED** [review]
LGTM

File: portfolio/src/main/webapp/script.js (hunk 2 above)
Comment (mariom3) [reply portfolio/src/main/webapp/script.js:40]: Done 265b9c0

**Pull request merged**